### DISCUSSION TITLE
How about word beacon, replace bmacro

### DISCUSSION THREAD

**Discussion by DogLooksGood** [meow-edit/meow#92] [General]
When bmacro(batched kmacro) state is introduced, I could find a good name for it. I picked the name just because it's 6 characters like all other names(insert, normal, motion, keypad).

What we actually do in bmacro state, is set markers(via fake cursors/regions) at all the places where later we want apply macro to. So what about the word __beacon__?

**Comment by VitalyAnkh** [comment]
Looks good to me.
bmacro and kmacro are so similar and they confuse people, in my opinion.

**Reply by DogLooksGood** [reply]
Looks like the change will only affect the value of valiable `meow-replace-state-name-list`.

Users won't be affected and I'll make the change.

Please use "beacon state" in tutor.

**Comment by DogLooksGood** [comment]
changed in 16cd220483a9f0475411493d10370d093299f561.

**Comment by mclearc** [comment]
Just FYI -- there is already a package named "beacon": https://github.com/Malabarba/beacon. I also don't see any either mnemonic or portmanteau advantage to calling the macro state "beacon". Are you willing to reconsider the name?

**Reply by DogLooksGood** [reply]
Probably not, but any good suggestion?

Although meow beacon state and beacon are entirely different, but they all used to create beacons which remind user for the places in buffer.

---

The renaming will be breaking change, even though it just affect one user customize variable.

**Reply by mclearc** [reply]
Yeah, good names are difficult! Here are some random thoughts for the mode name, but I understand if you don't want to keep changing the name.

- batch
- multi
- multiedit
- medit
- macedit
- mcedit
- emacro ("edit macro")
- cursors
- mcursor
- plural
- mancur ("many cursors")

**Reply by DogLooksGood** [reply]
Most names in the list were in my consideration.

But I want a name, that
- must be 6 characters long, like all other names
- must be not start with n, i, k,p. for we can represent each state in one character
- must be a real word, not scrabble words(why I don't like bmacro)

I like `plural`, sounds good. But `beacon` is also a good name in my opinion.
Here beancon stands for `fake cursor and regions where later we call keyboard macro`.

**Reply by smarticle-particle** [reply]
I know I’m a bit late, but I thought of another name idea: ```shadow```. The fake cursors are like shadows of the original cursor. (Or if you like Naruto, they are like shadow clones.)
